lib/jobs/dell-wsman-add-hotspare-updatexml.js

```diff
@@ -0,0 +1,162 @@
+    BaseJob,
+    Logger,
+    assert,
+    util,
+    configuration,
+    errors,
+    Smb2Client,
+    NfsClient,
+    Promise,
+    _
+) {
+    var logger = Logger.initialize(WsmanAddHotspareUpdateXmlFactory);
+
+    function WsmanAddHotspareUpdateXmlJob(options, context, taskId) {
+        WsmanAddHotspareUpdateXmlJob.super_.call(this, logger, options, context, taskId);
+    }
+
+    util.inherits(WsmanAddHotspareUpdateXmlJob, BaseJob);
+
+    WsmanAddHotspareUpdateXmlJob.prototype._run = function() {
+        var self = this;
+        if(!self.options.driveId) {
+            throw new errors.NotFoundError('The driveId should not be empty.');
+        }
+        self.dell = configuration.get('dell');
+        if(!self.dell || !self.dell.shareFolder) {
+            throw new errors.NotFoundError('Share folder is not defined in smiConfig.json');
+        }
+        return self.updateXmlForRaid();
+    };
+
+    WsmanAddHotspareUpdateXmlJob.prototype.updateXmlForRaid = function() {
+        var self = this;
+        var fileName = self.context.graphId + '.xml';
+        var client;
+        return Promise.try(function() {
+            if(self.dell.shareFolder.shareType === 0) {
+                client = new NfsClient(
+                    self.dell.shareFolder.address,
+                    self.dell.shareFolder.shareName,
+                    self.context.mountDir
+                );
+            } else if(self.dell.shareFolder.shareType === 2) {
+                client = new Smb2Client(
+                    self.dell.shareFolder.address,
+                    self.dell.shareFolder.shareName,
+                    self.dell.shareFolder.username,
+                    self.dell.shareFolder.password
+                );
+            } else {
+                throw new Error('Invalid shareType in smiConfig.json. The shareType should be 0 or 2.');
+            }
+            return client.readFile(fileName);
+        })
+        .then(function(data) {
+            return self.updateXml(data);
+        })
+        .then(function(doc) {
+            return client.writeFile(fileName, doc);
+        })
+        .then(function() {
+            self._done();
+        })
+        .catch(function(err) {
+            logger.error('An error occurred while updating component xml.', { error: err });
+            self._done(err);
+        });
+    };
+
+    WsmanAddHotspareUpdateXmlJob.prototype.updateXml = function(data) {
+        var self = this;
+        var xmlFile = String.fromCharCode.apply(null, new Uint16Array(data));
+        var doc = new xmldom().parseFromString(xmlFile, 'application/xml');
+        if(self.options.hotspareType === 'dhs') {
+            self.UpdateVolumeElement(doc);
+        } else {
+            self.UpdateDiskElement(doc);
+        }
+        return doc;
+    };
+
+    /*
+     * Dedicated hotspare should be added into corresponding volume
+     */
+    WsmanAddHotspareUpdateXmlJob.prototype.UpdateVolumeElement = function(doc) {
```

Comment:
naming style suggestion: the PascalCase style function name is usually used for constructer or class, for function, use camelCase instead.